### ISSUE TITLE
[Accessibility]: Announcement for collapsible sections in ref panel is hard to understand

### ISSUE THREAD

**Issue by varungandhi-src** [sourcegraph/sourcegraph#36576]
### Audit type

Screen reader navigation

### User journey audit issue

https://github.com/sourcegraph/sourcegraph/issues/33510

### Problem description

Right now, several expanded collapsible sections are announced as "Close thing, expanded, button, group". This seems hard to understand; it seems like the "Close" is part of the button.

### Expected behavior

It would be nice to announce the sections as "thing, click to collapse this section" or similar.

### Additional details

![image](https://user-images.githubusercontent.com/93103176/171944795-5e5fcd4b-dea6-4fcd-98f2-cc4319999603.png)

([example URL](https://sourcegraph.com/github.com/sourcegraph/sourcegraph/-/blob/cmd/server/internal/goreman/log.go?L23:5&subtree=true#tab=references))

### Assigning labels

- [ ] Please give this issue an estimate by applying a label like `estimate/Xd`, where X is the estimated number of days it will take to complete.
- [ ] If this issue is specific to a specific Sourcegraph product, please assign the appropriate team label to this issue.
- [ ] If this issue will require input from designers in order to complete, please assign the label `needs-design`.
- [ ] If you are confident that this issue should be fixed by GitStart, please assign the label `gitstart`.

### Owner

I'm unsure who should work on this issue or if it needs more clarification. I will let the Frontend Platform team triage this.
